assets/js/plugin-check-admin.js

```diff
@@ -172,14 +172,41 @@
 	 * @param {Object} data The response data.
 	 */
 	async function runChecks( data ) {
+		let isSuccessMessage = true;
+		let messageClass = '';
+		let messageText = '';
 		for ( let i = 0; i < data.checks.length; i++ ) {
 			try {
 				const results = await runCheck( data.plugin, data.checks[ i ] );
+				const errorsLength = Object.values( results.errors ).length;
+				const warningsLength = Object.values( results.warnings ).length;
+				if (
+					isSuccessMessage &&
+					( errorsLength > 0 || warningsLength > 0 )
+				) {
+					isSuccessMessage = false;
+				}
 				renderResults( results );
 			} catch ( e ) {
 				// Ignore for now.
 			}
 		}
+
+		if ( isSuccessMessage ) {
+			messageClass = 'notice-success';
+			messageText = pluginCheck.successMessage;
+		} else {
+			messageClass = 'notice-error';
+			messageText = pluginCheck.errorMessage;
+		}
+
+		resultsContainer.innerHTML += renderTemplate(
+			'plugin-check-results-complete',
+			{
+				class: messageClass,
```

Comment:
The `class` name is somewhat problematic to use since it is also a language keyword. I'd prefer that we use something else like `className`.

Or alternatively, we change this to a more general `type` value, to which we just pass "success" or "error". If we do that, then the template itself would need to use `notice-{{ data.type }}`.

Reply:
Make sense. Thanks for pointing it out. Address in https://github.com/10up/plugin-check/pull/160/commits/8d6c539c8824f17777e44c14c98a01d55cb25362